tests/semanticdb/metac.expect

```diff
@@ -3302,6 +3380,15 @@ Occurrences:
 [39:10..39:17): leftVar -> local3
 [40:10..40:18): rightVar -> local4
 
+Synthetics:
```

Comment:
Unlike [scala2](https://github.com/scalameta/scalameta/blob/179207a9f8dc0f5a51c2861521feba1473f725d5/tests/jvm/src/test/resources/metac.expect#L4782-L4784), we don't have `[4:6..4:19): (left, right) => *.apply[Int, Int]` because actually we don't have it.

`-Xprint:typer` shows:

before
```scala
  val (left, right) = (1, 2)
```

after
```scala
    private[this] val $1$: (Int, Int) = Tuple2.apply[Int, Int](1, 2)
    val left: Int = this.$1$._1
    val right: Int = this.$1$._2
```

Reply:
I didn't add synthesized `Tuple2.apply[Int, Int](*)` to Synthetic section, because Scala2 doesn't include it (even though there is synthesized `Tuple2.apply` )

Reply:
maybe something can be done here with tree attachments, this case only occurs when a pattern is exactly identifiers in a tuple:

https://github.com/dotty-staging/dotty/blob/82677dab26eb2c9bd6c978f6bbf4d5318d797cc6/compiler/src/dotty/tools/dotc/ast/Desugar.scala#L1126-L1128

Reply:
or at least the code linked above deals with all desugaring of patterns in val defs so the door is open for other patterns than tuples